persistence/db.go

```diff
@@ -129,17 +140,126 @@ func InitializeAccountTables(ctx context.Context, db *pgx.Conn) error {
+
+	addValueToPool := func(poolName string, valueToAdd string) error {
+		value, err := types.StringToBigInt(valueToAdd)
+		if err != nil {
+			return err
+		}
+		if present := poolValues[poolName]; present == nil {
+			poolValues[poolName] = big.NewInt(0)
+		}
+		poolValues[poolName].Add(poolValues[poolName], value)
+		return nil
+	}
+
+	for _, v := range state.Validators {
+		// DISCUSS_IN_THIS_COMMIT(drewskey): If the validator (and other actors) have a balance, do we expect that
+		// to be in the `accounts` type corresponding to the same address?
+		if err := ctx.SetAccountAmount(v.Address, "0"); err != nil {
+			return err
+		}
+		if err := ctx.InsertValidator(v.Address, v.PublicKey, v.Output, v.Paused, int(v.Status), v.ServiceUrl, v.StakedTokens, v.PausedHeight, v.UnstakingHeight); err != nil {
+			return err
+		}
+		if err := addValueToPool(genesis.ValidatorStakePoolName, v.StakedTokens); err != nil {
+			return err
+		}
+	}
+
+	for _, f := range state.Fishermen {
+		if err := ctx.InsertFisherman(f.Address, f.PublicKey, f.Output, f.Paused, int(f.Status), f.ServiceUrl, f.StakedTokens, f.Chains, f.PausedHeight, f.UnstakingHeight); err != nil {
+			return err
+		}
+		if err := ctx.SetAccountAmount(f.Address, "0"); err != nil {
+			return err
+		}
+		if err := addValueToPool(genesis.FishermanStakePoolName, f.StakedTokens); err != nil {
+			return err
+		}
+	}
+
+	for _, sn := range state.ServiceNodes {
+		if err := ctx.InsertServiceNode(sn.Address, sn.PublicKey, sn.Output, sn.Paused, int(sn.Status), sn.ServiceUrl, sn.StakedTokens, sn.Chains, sn.PausedHeight, sn.UnstakingHeight); err != nil {
+			return err
+		}
+		if err := ctx.SetAccountAmount(sn.Address, "0"); err != nil {
+			return err
+		}
+		if err := addValueToPool(genesis.ServiceNodeStakePoolName, sn.StakedTokens); err != nil {
+			return err
+		}
+	}
+
+	for _, app := range state.Apps {
+		if err := ctx.InsertApp(app.Address, app.PublicKey, app.Output, app.Paused, int(app.Status), app.MaxRelays, app.StakedTokens, app.Chains, app.PausedHeight, app.UnstakingHeight); err != nil {
+			return err
+		}
+		if err := ctx.SetAccountAmount(app.Address, "0"); err != nil {
+			return err
+		}
+		if err := addValueToPool(genesis.AppStakePoolName, app.StakedTokens); err != nil {
+			return err
+		}
+	}
+
+	for _, acc := range state.Accounts {
+		if err := ctx.AddAccountAmount(acc.Address, acc.Amount); err != nil {
+			return err
+		}
+	}
+
+	for _, pool := range state.Pools {
+		if err := ctx.InsertPool(pool.Name, pool.Account.Address, pool.Account.Amount); err != nil {
+			return err
+		}
+		// DISCUSS_IN_THIS_COMMIT(drewskey): What if there's a discrepency between `pool.Account.Amount` and `poolValues`?
+		if err := ctx.SetAccountAmount(pool.Account.Address, pool.Account.Amount); err != nil {
+			return err
+		}
+	}
+
+	if err := ctx.InitParams(); err != nil {
+		return err
+	}
+	// TECHDEBT: Calling this for each param is not practical. Need someone to refactor the gov params table.
```

Comment:
Updated the comment to this:
```
	// TECHDEBT: Calling the setter (e.g. SetValidatorMaximumMissedBlocks, Set_) for this for each param results in a lot
	// of code bloat and is not practical. This will need to be fixed as part of the gov param refactor task.
```

Lmk if that's clearer.